eth/vm/code_stream.py

```diff
@@ -25,34 +25,34 @@ class CodeStream(object):
 
     def __init__(self, code_bytes: bytes) -> None:
         validate_is_bytes(code_bytes, title="CodeStream bytes")
-        self.stream = io.BytesIO(code_bytes)
+        stream = io.BytesIO(code_bytes)
+        self.stream = stream
+        self._bound_stream_read = stream.read
         self._raw_code_bytes = code_bytes
         self._length_cache = len(code_bytes)
         self.invalid_positions = set()
         self.valid_positions = set()
 
     def read(self, size: int) -> bytes:
-        return self.stream.read(size)
+        return self._bound_stream_read(size)
 
     def __len__(self) -> int:
         return self._length_cache
 
     def __iter__(self) -> 'CodeStream':
         return self
 
-    def __next__(self) -> int:
-        return self._next()
-
     def __getitem__(self, i: int) -> int:
         return self._raw_code_bytes[i]
 
-    def _next(self) -> int:
-        next_opcode_as_byte = self.read(1)
+    def __next__(self) -> int:
+        # a very performance-sensitive method
+        next_opcode_as_byte = self._bound_stream_read(1)
 
         if next_opcode_as_byte:
```

Comment:
In theory if this was done as a generator function we could do something like this to get rid of this `if` statement.

```python
def __iter__(self) -> Generator[int, None, None]:
    read = self.stream.read
    try:
        while True:
            yield ord(read(1))
    except TypeError?  # whatever `ord(b'')` raises
        yield STOP
```